src/reanimated2/animations.js

```diff
@@ -1,37 +1,262 @@
 import NativeReanimated from './NativeReanimated';
 
 let IN_STYLE_UPDATER = false;
 
-const assertNumber = (value, callerName) => {
-  'worklet';
-  const valueType = typeof value;
-  if (valueType !== 'number') {
-    let error = `invalid type of toValue passed to ${callerName}, expected \`number\`, got \`${valueType}\``;
-    if (valueType === 'object') {
-      error += ', maybe you forgot to add `.value`?';
-    }
-    throw new Error(error);
-  }
-};
-
 export function initialUpdaterRun(updater) {
   IN_STYLE_UPDATER = true;
   const result = updater();
   IN_STYLE_UPDATER = false;
   return result;
 }
 
+/* accepts parameters
+ * r  Object = {r:x, g:y, b:z}
+ * OR 
+ * r, g, b
+ */
+function RGBtoHSV(r, g, b) {
+  'worklet';
+  if (arguments.length === 1) {
+    g = r.g, b = r.b, r = r.r;
+  }
+  var max = Math.max(r, g, b),
+    min = Math.min(r, g, b),
+    d = max - min,
+    h,
+    s = (max === 0 ? 0 : d / max),
+    v = max / 255;
+
+  switch (max) {
+    case min:
+      h = 0;
+      break;
+    case r:
+      h = (g - b) + d * (g < b ? 6 : 0);
+      h /= 6 * d;
+      break;
+    case g:
+      h = (b - r) + d * 2;
+      h /= 6 * d;
+      break;
+    case b:
+      h = (r - g) + d * 4;
+      h /= 6 * d;
+      break;
+  }
+
+  return {
+    h: h,
+    s: s,
+    v: v
+  };
+}
+
+/* accepts parameters
+ * h  Object = {h:x, s:y, v:z}
+ * OR 
+ * h, s, v
+ */
+function HSVtoRGB(h, s, v) {
+  'worklet';
+  var r, g, b, i, f, p, q, t;
+  if (arguments.length === 1) {
+    s = h.s, v = h.v, h = h.h;
+  }
+  i = Math.floor(h * 6);
+  f = h * 6 - i;
+  p = v * (1 - s);
+  q = v * (1 - f * s);
+  t = v * (1 - (1 - f) * s);
+  switch (i % 6) {
+    case 0:
+      r = v, g = t, b = p;
+      break;
+    case 1:
+      r = q, g = v, b = p;
+      break;
+    case 2:
+      r = p, g = v, b = t;
+      break;
+    case 3:
+      r = p, g = q, b = v;
+      break;
+    case 4:
+      r = t, g = p, b = v;
+      break;
+    case 5:
+      r = v, g = p, b = q;
+      break;
+  }
+  return {
+    r: Math.round(r * 255),
+    g: Math.round(g * 255),
+    b: Math.round(b * 255)
+  };
+}
+
+export function transform(value, handler) {
+  'worklet';
+  if (value === undefined) {
+    return undefined;
+  }
+
+  if (typeof value === 'string') {
+    // toInt
+    // TODO handle color
+    const match = value.match(/(\D*)([\d.]*)(\D*)/);
+    const prefix = match[1];
+    const suffix = match[3];
+    const number = match[2];
+    handler.__prefix = prefix;
+    handler.__suffix = suffix;
+    return parseFloat(number);
+  }
+  
+  // toString if __prefix is available and number otherwise
+  if (handler.__prefix === undefined) {
+    return value;
+  }
+
+  return handler.__prefix + value + handler.__suffix;
+}
+
+const needsToBeChanged = ['toValue', 'current', 'startValue'];
+export function transformAnimation(animation) {
+  'worklet';
+  if (!animation) {
+    return;
+  }
+  needsToBeChanged.forEach((prop) => {
```

Comment:
I commented in the other place that it may not be worth transforming all the values on each frame. I'm not a fan of loops like this here and would prefer to just have three lines:
```
animation.toValue = transform(animation.toValue, animation);
animation.current = transform(animation.current, animation);
animation.startValue = transform(animation.startValue, animation);
```
which is also more efficient (can actually be jitted and saves some allocations). But that isn't as much important as the fact that I think we don't really need to transform all those three props every frame.

Reply:
fixed ec76ca0